packages/oidc/src/token.ts

```diff
@@ -0,0 +1,226 @@
+interface TokenPayload {
+  sub: string;
+  name: string;
+  exp: number;
+}
+
+function isNodeEnv(): boolean {
+  return (
+    typeof process !== 'undefined' &&
+    process.versions != null &&
+    process.versions.node != null
+  );
+}
+
+function isBrowserEnv(): boolean {
+  return typeof window !== 'undefined';
+}
+
+async function getVercelDataDir(): Promise<string | null> {
+  if (isBrowserEnv()) {
+    return '';
+  }
+  const vercelFolder = 'com.vercel.cli';
+  const os = await import('os');
+  const path = await import('path');
+  if (process.env.XDG_DATA_HOME) {
+    return path.join(process.env.XDG_DATA_HOME, vercelFolder);
+  }
+  switch (os.platform()) {
+    case 'darwin':
+      return path.join('~/Library/Application Support', vercelFolder);
+    case 'linux':
+      return path.join('~/.local/share', vercelFolder);
+    case 'win32':
+      return path.join('%LOCALAPPDATA%', vercelFolder);
+    default:
+      return null;
+  }
+}
+
+async function getVercelCliToken(): Promise<string | null> {
+  const dataDir = await getVercelDataDir();
+  if (!dataDir) {
+    return null;
+  }
+  const path = await import('path');
+  const fs = await import('fs');
```

Comment:
```suggestion
  const [path, fs] = await Promise.all([
    import('path'),
    import('fs'),
  ]);
```
Maybe also better to move to top of function declaration, before the `getVercelDataDir()`?